src/components/Navigator.js

```diff
@@ -3,74 +3,141 @@ import { Subject } from 'rxjs';
 let createNavigator = () => {
   let subject = new Subject();
 
+  let possibleMoves = [];
+  let possibleDirections = {};
+
   /**
-   * Returns possible directions from state on each level.
-   * If we are at state [1, 0] depending on the map, the
-   * result might look something like this:
-   * [
-   *    {
-   *      next: [2],
-   *      previous: [0]
-   *    },
-   *    {
-   *      next: [1, 1],
-   *      previous: false
-   *    }
-   * ]
+   * Maps motion-names to position in directions array
    *
-   * @param {number[]} state State for which to return the directions
-   * @param {*[]} elements Elements to go through
-   * @param {number} level Depth of recursion. Defaults to 0
-   * @return {[directions]} List of direcitons object
+   * @type {{
+   *  left: {level: number, direction: string},
+   *  up: {level: number, direction: string},
+   *  right: {level: number, direction: string},
+   *  down: {level: number, direction: string}
+   * }}
    */
-  let getDirections = (state, elements, level = 0) => {
-    if (level  === state.length) return [];
+  let motions = {
```

Comment:
I can add this in now, but would prefer to do that in an own step. We're just refactoring at the moment. But I created a new issue for it: #37